packages/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -702,6 +723,17 @@ export const useGridColumnResize = (
         }
 
         apiRef.current.updateColumns(newColumns);
+
+        if (colDefRef.current) {
+          const columnsState = gridColumnsStateSelector(apiRef.current.state);
+          const column: GridStateColDef = columnsState.lookup[colDefRef.current.field];
+          const width: number = column.width as number;
+          apiRef.current.publishEvent('columnWidthChange', {
+            element: apiRef.current.getColumnHeaderElement(colDefRef.current.field),
+            colDef: { ...colDefRef.current, width },
+            width,
+          });
+        }
```

Comment:
This is incorrect. You have the column definitions in `newColumns`, you shouldn't use the one stored in that ref. You should fire the event for each column in `newColumns` for which the width has changed.

Reply:
newColumns is an array should I trigger this event for only one column for which the autoResize is done or each of the element of the array.

```
if (colDefRef.current) {
          const updatedColumn: GridStateColDef | undefined = newColumns.find((newColumn) => {
            return newColumn.field === colDefRef.current?.field;
          });

          if (updatedColumn) {
            const width: number = updatedColumn.width as number;
            apiRef.current.publishEvent('columnWidthChange', {
              element: apiRef.current.getColumnHeaderElement(colDefRef.current.field),
              colDef: updatedColumn,
              width,
            });
          }
        }
```

Reply:
The `autosizeColumns` can operate on multiple columns, not just one. The `if` guard here does not apply, `colDefRef` belongs to the resize logic. We need to trigger the event for each column that changed width.

Reply:
Done